addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -644,9 +644,15 @@ export const editorCommands = {
                     } else {
                         // No <font> found: insert a new one.
                         font = document.createElement('font');
-                        node.after(font);
+                        // Insert the font inside the node if it's a span with font size.
+                        if (node.tagName === 'SPAN' && node.style['fontSize']) {
+                            node.append(font);
+                        } else {
+                            node.after(font);
+                        }
                     }
-                    if (node.textContent) {
+                    // Move the node inside the <font> when it's not a link, not a span with font size and with text content.
+                    if (node.textContent && node.nodeName !== 'A' && !(node.nodeName === 'SPAN' && node.style['fontSize'])) {
                         font.appendChild(node);
                     } else {
```

Comment:
Since you change the `if` condition, all `<a>` element and all span with `fontSize` style will land here and trigger a `fillEmpty`.

I don't think this is want we want. Or maybe I don't understand the change well enough. 
In any case, could you explain to me why it is like this ?

Reply:
Thanks for checking!, so I wanted to make sure when it's a link node or a span with font size, we put the `<font>` node inside of the link or the span. For now, we put the span and link inside of the `<font>` by using `appendChild`. This causes issues when:
1. the font size is set by a span, the background color won't match the font size (because the font node is inside of the span)
2. there are more than three link list items, the middle ones are not colored (because the `selectedNodes` has the link nodes `<a>` and we try to put the link inside of a font)

While writing the reasons why I put these conditions, I think maybe a more reasonable way is to only keep the text nodes in the `selectedNodes`, so we always create the font nodes right next to the text nodes. But it will probably break some existing tests related to `<font>`

Lemme know what do you think :pray: